### DISCUSSION TITLE
is there any state management like in flutter for flet

### DISCUSSION THREAD

**Discussion by bobwatcherx** [flet-dev/flet#803] [Q&A]
I want to manage my state for several pages, is there any state management or anything to manage state thanks

**Comment by FeodorFitsner** [comment]
Does https://flet.dev/docs/guides/python/session-storage work?

**Comment by abcnow** [comment]
@FeodorFitsner Fantastic approach with the session-storage! Are they stored locally for that particular page and access globally as well?

**Reply by FeodorFitsner** [reply]
It's per session (aka user connection).

**Comment by abcnow** [comment]
Feodor.. after reading about Controls.. i think that's more likely to have a better similarity to flutter's state management .. don't you think? You even have the Counter App there ->by the way ... I think it could be simpler.. on one end.. by default it could update itself so that we don't have to keep writing self.update.. on the other end.. do we really need to use "self." every time? Can we just use "self." the first time ->see a crazy possible proposition at the end. .

import flet as ft

class Counter(ft.UserControl):
    def add_click(self, e):
        self.counter += 1
        self.text.value = str(self.counter)
        self.update()

    def build(self):
        self.counter = 0
        self.text = ft.Text(str(self.counter))
        return ft.Row([self.text, ft.ElevatedButton("Add", on_click=self.add_click)])

def main(page):
    page.add(Counter(), Counter())

ft.app(target=main)

------ crazy possible proposition..

class Counter(ft.UserControl):
    def add_click(self@, e):
        @counter += 1 (0) // where 0 is initial value
        return = ft.Row(@text from @counter, ft.ElevatedButton("Add", on_click=@add_click))

**Reply by FeodorFitsner** [reply]
What are those variables starting from `@`? 🧐

**Reply by abcnow** [reply]
An attempt to replace "self." With 1 character "@" but it t could be something else....

**Reply by ndonkoHenri** [reply]
Can python parse that? 😄

**Comment by iak-a-dev** [comment]
Would there be any other options, such as Bloc or Riverpod ?
